driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -1593,15 +1593,15 @@ private void connect() {
                 if (serverResponseVersion == attemptedVersion) {
                     // We're post CASSANDRA-11464: the server replies with the same version that the client tried (to
                     // ensure that the client can decode the response).
-                    // Retry with a lower version (if it's too low we'll eventually reach our min and fail).
+                    // Retry with a lower version (if it's too low we'll fail at the next attempt, as we only retry once).
```

Comment:
> This is getting complicated. How about going back to the simple algorithm used before JAVA-361.

I don't have anything against the idea. I will push the modifications shortly if nobody disagrees.